### PR TITLE
Feature/compound integration

### PR DESCRIPTION
**Pull request by pRoy24**
Added the fully functional feature and fixed the rounds of comments in https://github.com/protofire/omen-exchange/pull/1466

- [x] Test token with cToken collateral pool creation.
- [x] Test outcome token buy for token with cToken collateral by buying via base token.
- [x] Test outcome token buy for token with cToken collateral by buying via cToken.
- [x] Test outcome token sell for token with cToken collateral by withdrawing as base token
- [x] Test outcome token sell for token with cToken collateral by withdrawing as cToken.
- [x] Test pool token buy for token with cToken collateral adding base token.
- [x] Test pool token buy for token with cToken collateral adding cToken.
- [x] Test pool token sell for token with cToken collateral by withdrawing as base token.
- [x] Test pool token sell for token with cToken collateral by withdrawing as cToken.
- [x] On changing currency filters on all of the above flows the prices should show in the corresponding currency.
- [x] All prices and values for tokens/liquidity/volume with cToken collateral should be scaled up in terms of the base collateral prices.
- [x] Test token without pool token flows for any regression.
- [x] Test special case ETH->WETH pool for any regression.
- [x] Test special case ETH -> cETH pool for all the pool flows.
- [x] cToken pools should display interest earning pool badge.
- [x] Verify that the APY% if showing and consistent with the values for the corresponding token in app.compound.finance.
- [x] Test that the cToken feature is disabled and there is no regression in xDai and Sokol networks.
- [x] Test all the above flows for closed markets as well.
- [x] Verify that the pool creation Checbox and "Earn interest" display and the cToken pool badge is consistent with the Figma mockups.

### CODE REVIEW

**Review by hexyls — CHANGES_REQUESTED** [review]

File: app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx
Comment (pRoy24) [reply app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx]: Yes sounds good, consolidated that function into tools

File: app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx
Comment (pRoy24) [reply app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx:409]: Good catch there was a decimals bug in the withdraw pool token as base token flow which was causing the amount to sell to be 0 every time.

File: app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx
Comment (hexyls) [inline app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx:244]: atm if I create a cDai market with 1 Dai worth of liquidity it shows 0 pool tokens:

Screen Shot 2021-01-27 at 11 09 41 am

File: app/src/services/cpk.ts
Comment (hexyls) [inline app/src/services/cpk.ts:388]: we need a hardcoded gas value for native asset transactions, it looks like market creation with compound integration costs around 1,300,000 gas so a hardcode of around 1,500,000 gas would be great!

File: app/src/util/networks.ts
Comment (hexyls) [inline app/src/util/networks.ts:408]: adding ETH here adds a second ETH to the token dropdown menu if we haven't loaded the token list from the subgraph list yet:

Screen Shot 2021-01-27 at 11 27 46 am

File: app/src/components/market/sections/market_create/market_wizard_creator_container.tsx
Comment (hexyls) [inline app/src/components/market/sections/market_create/market_wizard_creator_container.tsx:44]: Just checking - are we planning on adding compound support for scalar markets later on? Sounds good to me if so.

**Review by hexyls — CHANGES_REQUESTED** [review]

File: app/src/components/market/sections/market_create/market_wizard_creator_container.tsx
Comment (pRoy24) [reply app/src/components/market/sections/market_create/market_wizard_creator_container.tsx:44]: Yes we do after this gets merged

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Apy part sticks out....it should at least wrap to new row...
Screenshot 2021-01-27 at 14 45 06
-I would suggest different way of handling the wrapping of elements when they overstretch..the way they are right now leaves blank space on the right instead of being evenly distributed...I would add flexbox that wraps to new row when items don't fit...what do you think @pimato 
Screenshot 2021-01-27 at 14 57 28

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Don't know if it's related to this pr but when i have DAI selected it asks me for permission to spend cDai...
Screenshot 2021-01-27 at 14 53 19

File: app/src/services/cpk.ts
Comment (pRoy24) [reply app/src/services/cpk.ts:388]: Added gas value hardcode

**Comment by pimato** [comment]
> Apy part sticks out....it should at least wrap to new row...
> Screenshot 2021-01-27 at 14 45 06
> -I would suggest different way of handling the wrapping of elements when they overstretch..the way they are right now leaves blank space on the right instead of being evenly distributed...I would add flexbox that wraps to new row when items don't fit...what do you think @pimato
> Screenshot 2021-01-27 at 14 57 28

correct!

File: app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx
Comment (pRoy24) [reply app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx]: Couldn't reproduce this on my last few attempts, Should be fixed now.

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Screenshot 2021-02-03 at 23 23 10

I would maybe add justify-content:space-between; it makes it look nicer like that...But seems to create issue with tooltips then?....
Below is how it is currently it just delets whitespace
Screenshot 2021-02-03 at 23 22 57

**Comment by pRoy24** [comment]
> Screenshot 2021-02-03 at 23 23 10
> 
> I would maybe add justify-content:space-between; it makes it look nicer like that...But seems to create issue with tooltips then?....
> Below is how it is currently it just delets whitespace
> Screenshot 2021-02-03 at 23 22 57

Discussed on the omen group chat and decided that best to keep it the way it is currently

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
- I'm still getting cDai setAllowance for DAI market...
Screenshot 2021-02-08 at 12 49 46
- No margin between columns in the APY tooltip sections
Screenshot 2021-02-08 at 12 51 44

**Review by Mi-Lan — APPROVED** [review]
Nicely done!

**Review by hexyls — CHANGES_REQUESTED** [review]
Looks great man! I found one more issue but imo it looks good to go after that 🙌

When I go to withdraw liquidity it works perfectly at first. If I swap to withdrawing cDai and change the pool token amount the numbers go a little wonky, check out the recording:

https://user-images.githubusercontent.com/72715177/107297342-3a0e8b80-6ac7-11eb-810f-90ae2ee1c4bc.mov

**Comment by pRoy24** [comment]
> Looks great man! I found one more issue but imo it looks good to go after that 🙌
> 
> When I go to withdraw liquidity it works perfectly at first. If I swap to withdrawing cDai and change the pool token amount the numbers go a little wonky, check out the recording:
> 
>  Screen.Recording.2021-02-09.at.10.42.48.am.mov

Thanks good catch 👍 , this seems to have crept in during one of the rebases.
Fixed in latest push.

**Comment by KadenZipfel** [comment]
Getting the following when trying to create a scalar market
image

**Comment by pRoy24** [comment]
> Getting the following when trying to create a scalar market
> image

It hasn't been tested on Scalar market so the option shouldn't be visible for scalar market creation.

Having said that, it should work with scalar markets as well with minimal changes.
Disabled the "Use c token" option when creating scalar markets for now, will enable this and work on scalar market integration as part of https://github.com/protofire/omen-exchange/issues/1637

**Review by hexyls — APPROVED** [review]

**Comment by pimato** [comment]
[](https://gifyu.com/image/VmuJ)

the onHold styling removes the border

**Comment by pimato** [comment]
checking on Rinkeby and it seems like after I click create market, I am getting:
Bildschirmfoto 2021-02-10 um 15 11 56

**Comment by pimato** [comment]
Should be align to the right:
Bildschirmfoto 2021-02-10 um 15 17 21

**Review by pimato — CHANGES_REQUESTED** [review]
see comments

**Review by pimato — APPROVED** [review]

**Pull request merged**
